### PR TITLE
optparse: add flags fields to option and subcommand structures

### PR DESCRIPTION
**Pull request by grondo**
As suggested by @chu11, add new `flags` fields to `struct optparse_option` and `struct optparse_subcommand` now, before there are many more call sites to update.

The `struct optparse_option` change was actually very painless, since almost all callers used designated initializers, most use cases did not even have to change. I did, however, remove the special case `list_argument` (i.e. `has_arg == 3`) extension and instead made this an option flag `OPTPARSE_OPT_AUTOSPLIT`, which is a nice improvement over the previous, more messy approach.

The addition of subcommand flags was a bit more work, and is perhaps questionable since we already had a way to set arbitrary options/flags/attributes of a subcommand with `optparse_set (p, ITEM, ...)`. However it did seem a nice improvement to keep subcommand initializer all together in one place when using an `optparse_subcommand` table, so I went ahead and did it. I'd appreciate feedback if this was the right decision.

A flag `OPTPARSE_SUBCMD_SKIP_OPTS` was added that does the same thing as `optparse_set (subcmd, OPTPARSE_SUBCMD_NOOPTS, 1)` and the one caller (`flux hwloc lstopo`) was updated to use the flag initializer instead of `optparse_set`.

In both cases, a `*_HIDDEN` option was added, since that was a trivial extension and will now be there if we need it.

### CODE REVIEW

**Comment by chu11** [comment]
I like the change to	```has_arg```, it appears to be	much clearer now.  Outside of the few nits I noticed, I saw you added a OPTPARSE_SUBCMD_HIDDEN test but not OPTPARSE_SUBCMD_HIDE.  Should we test both?

**Comment by grondo** [comment]
Yeah, more tests should be added, but I wanted feedback on the subcommand changes before cleaning this up as a meege candidate. 

Am I just missing the other review notes, or are those in progress?

File: src/cmd/builtin/content.c [hunk 1]
```diff
@@ -210,30 +210,34 @@ static struct optparse_subcommand content_subcmds[] = {
...
       internal_content_store,
+      0,
       store_opts,
     },
     { "dropcache",
       NULL,
       "Drop non-essential entries from local content cache",
       internal_content_dropcache,
-      NULL,
+      0, NULL,
```
Comment (chu11) [inline src/cmd/builtin/content.c]: nit formatting, should be a separate line like others in this file

**Comment by chu11** [comment]
@grondo Ahh, I needed to click the "submit review" button on the "Files Changed" tab.  Dunno if that's new or maybe how you guys have things setup.  Hopefully you can see the two nits now.

File: src/common/liboptparse/optparse.h [hunk 2]
```diff
@@ -146,7 +158,7 @@ optparse_t *optparse_get_parent (optparse_t *p);
 
 /*
  *   Convenience function like optparse_add_subcommand, additionally
- *    registering with usage string [usage], documentation blurb [doc] and
+ *    registering with usage string [usage], documentation blurb[doc] and
```
Comment (grondo) [inline src/common/liboptparse/optparse.h]: Yes, thanks. Accidental commit

**Comment by garlick** [comment]
This looks good to me.  Rebase on current master and then merge?

**Comment by grondo** [comment]
Yeah, let me check it over once more and rebase

**Comment by grondo** [comment]
Ok, I threw on one more unrelated fix, a typo in `codecov.yml` for libjson-c was causing that library to be counted in codecov coverage check.

**Comment by chu11** [comment]
LGTM, +1

**Comment by garlick** [comment]
Ready?

**Comment by grondo** [comment]
Yes.

**Pull request merged**